edgepi-thermocouple.js

```diff
@@ -1,55 +1,47 @@
 module.exports = function (RED) {
   const rpc = require("@edgepi-cloud/edgepi-rpc");
 
-  const initializeNode = (config) => {
-    const transport =
-      config.transport === "Network"
-        ? `tcp://${config.tcpAddress}:${config.tcpPort}`
-        : "ipc:///tmp/edgepi.pipe";
-
-    try {
-      const tc = new rpc.TcService(transport);
-      console.debug("Thermocouple node initialized on: ", transport);
-      node.status({ fill: "green", shape: "ring", text: "tc initialized" });
-      return tc;
-    } catch (error) {
-      console.error(error);
-      node.status({
-        fill: "red",
-        shape: "ring",
-        text: "Initialization error.",
-      });
-    }
-  };
-
-  const ThermocoupleNode = (config) => {
+  function ThermocoupleNode(config) {
     RED.nodes.createNode(this, config);
     const node = this;
     let output = config.output;
 
     initializeNode(config).then((tc) => {
       this.on("input", async function (msg, send, done) {
-        node.status({ fill: "green", shape: "dot", text: "input recieved" });
+        node.status({ fill: "green", shape: "dot", text: "input received" });
         try {
           output = msg.payload || output;
           let temps = await tc.singleSample();
-          // if (output === "cj-lin") {
-          //   msg.payload = temps;
-          // } else {
-          //   msg.payload = output === "cj" ? temps[0] : temps[1];
-          // }
-          if (output === "lin") {
-            msg.payload = temps;
-          }
+          msg.payload = temps[0];
```

Comment:
are you sure linearized temp is at `temps[0]` ?

Reply:
oh wait. it should be temps[1]. Fixed